core/profiles/src/main/java/software/amazon/awssdk/profiles/ProfileFileLocation.java

```diff
@@ -37,12 +37,37 @@
     private ProfileFileLocation() {
     }
 
+
+    /**
+     * Resolve the path for the configuration file, regardless of whether it exists or not
+     *
+     * @see #configurationFileLocation()
+     */
+    public static Path configurationFilePath() {
+        return resolveProfileFilePath(
+            ProfileFileSystemSetting.AWS_CONFIG_FILE.getStringValue()
+                                                    .orElse(Paths.get(ProfileFileLocation.userHomeDirectory(),
+                                                                      ".aws", "config").toString()));
```

Comment:
Nit: A lot of repetition of this. Can we centralize it?

Reply:
Good point - fixed!